core/src/main/java/hudson/TcpSlaveAgentListener.java

```diff
@@ -128,21 +129,23 @@ public int getAdvertisedPort() {
 
     /**
      * Gets the host name that we advertise protocol clients to connect to.
+     *
      * @since 2.198
      */
     public String getAdvertisedHost() {
         if (CLI_HOST_NAME != null) {
-          return CLI_HOST_NAME;
+            return CLI_HOST_NAME;
         }
         try {
-            return new URL(Jenkins.get().getRootUrl()).getHost();
-        } catch (MalformedURLException e) {
+            return new URI(Jenkins.get().getRootUrl()).toURL().getHost();
+        } catch (MalformedURLException | URISyntaxException e) {
```

Comment:
URI exposes a getHost() so there is there a need to convert to a `URL`?

```suggestion
            return new URI(Jenkins.get().getRootUrl()).getHost();
        } catch (URISyntaxException e) {
```